vm/src/builtins/bool.rs

```diff
@@ -166,6 +174,48 @@ impl PyBool {
     }
 }
 
+macro_rules! int_method {
+    ($method:ident) => {
+        AtomicCell::new(PyInt::as_number().$method.load().to_owned())
+    };
+}
+
+impl AsNumber for PyBool {
+    fn as_number() -> &'static PyNumberMethods {
+        static AS_NUMBER: Lazy<PyNumberMethods> = Lazy::new(|| PyNumberMethods {
+            add: int_method!(add),
+            subtract: int_method!(subtract),
+            multiply: int_method!(multiply),
+            remainder: int_method!(remainder),
+            divmod: int_method!(divmod),
+            power: int_method!(power),
+            negative: int_method!(negative),
+            positive: int_method!(positive),
+            absolute: int_method!(absolute),
+            boolean: int_method!(boolean),
+            invert: int_method!(invert),
+            lshift: int_method!(lshift),
+            rshift: int_method!(rshift),
```

Comment:
Not sure yet, but I think those fields in bool have to be `None`.
In my opinion, If that's not working with None, any inherited object of numbers can make similar problem.

Reply:
It seems `int` operations are needed for `bool` in CPython.
```
>>> (True + True) ** True
2
```

Reply:
Since `PyBool` is a subclass of `PyInt` and inherits its methods, so `(True + True) ** True` should somehow work without the above overriding in `as_number` I think.

Reply:
I am sorry to make confusion. I agree bool need int operations.
But I'd like to say manually filling `PyNumberMethods` is not the way to do. 
We can do that job only for builtin types written in Rust.
Making them `None` and finding a way to make it work will be helpful to solve the problem in general.

Reply:
Did not test but I think PyBool is working with int number methods via mro search in PyNumber::find_methods().

But Maybe we can override it in PyBool to have some optimize

Reply:
In my opinion, because we will not lookup PyNumberMethods via mro, we need to fill it as @xiaozhiyan suggested.
But I'd like to avoid to manually copy them like this. This is better to be handled during inheritance procedure.

Reply:
a proper inheritance procedure would be possible using #4645

Reply:
No, it wasn't. Because that is a static variable. We have no way to edit it later.

Reply:
@qingshi163 do you have any suggestion?

Reply:
I think we can inherit slots when initializing the subclass

Reply:
https://github.com/python/cpython/blob/71db5dbcd714b2e1297c43538188dd69715feb9a/Objects/typeobject.c#L8849-L8902

slots will be inherit if subclass does not override it.

when there is a change of slots on the type, cpython recusively update slots for all the subclasses.

Reply:
> slots will be inherit if subclass does not override it.

It is true in cpython but not here yet. That's the reason we always call `mro_find_map()` to find slots.